2 tests fail with OSError: [Errno 7] Argument list too long: 'git' on CentOS

complement to #5942 

<details>
<summary>2 failing tests</summary> 

```shell
======================================================================
FAIL: datalad.support.tests.test_annexrepo.test_files_split_exc(<class 'datalad.support.gitrepo.GitRepo'>,)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/tests/utils.py", line 764, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/tests/test_annexrepo.py", line 2349, in check_files_split_exc
    assert_not_in('too long', str(ecm.exception))
AssertionError: 'too long' unexpectedly found in "[Errno 7] Argument list too long: 'git'"


======================================================================
ERROR: datalad.support.tests.test_annexrepo.test_files_split_exc(<class 'datalad.support.annexrepo.AnnexRepo'>,)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/tests/utils.py", line 764, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/tests/test_annexrepo.py", line 2345, in check_files_split_exc
    r.add(files)
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/gitrepo.py", line 324, in _wrap_normalize_paths
    result = func(self, files_new, *args, **kwargs)
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/annexrepo.py", line 1569, in add
    return list(self.add_(
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/annexrepo.py", line 1630, in add_
    expected_additions = {
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/annexrepo.py", line 1630, in <dictcomp>
    expected_additions = {
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/annexrepo.py", line 1611, in _get_to_be_added_recs
    for r in super(AnnexRepo, self).add_(
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/gitrepo.py", line 1424, in add_
    add_out = self._call_git(
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/support/gitrepo.py", line 2134, in _call_git
    res = runner.run_on_filelist_chunks(
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/cmd.py", line 658, in run_on_filelist_chunks
    res = self.run(
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/cmd.py", line 379, in run
    results = event_loop.run_until_complete(
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/site-packages/datalad/cmd.py", line 102, in run_async_cmd
    transport, protocol = await proc
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/asyncio/base_events.py", line 1661, in subprocess_exec
    transport = await self._make_subprocess_transport(
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/asyncio/unix_events.py", line 197, in _make_subprocess_transport
    transp = _UnixSubprocessTransport(self, protocol, args, shell,
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/asyncio/base_subprocess.py", line 36, in __init__
    self._start(args=args, shell=shell, stdin=stdin, stdout=stdout,
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/asyncio/unix_events.py", line 789, in _start
    self._proc = subprocess.Popen(
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/dartfs-hpc/rc/home/v/d31548v/.conda/envs/datalad/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 7] Argument list too long: 'git'

```
</details>

so I guess we should review our consts for determining splitting into chunks

--- a/datalad/utils.py
+++ b/datalad/utils.py
@@ -108,9 +108,13 @@ CMD_MAX_ARG_HARDCODED = 2097152 if on_linux else 262144 if on_osx else 32767
 try:
     CMD_MAX_ARG = os.sysconf('SC_ARG_MAX')
     assert CMD_MAX_ARG > 0
-    if sys.version_info[:2] == (3, 4):
+    if CMD_MAX_ARG > CMD_MAX_ARG_HARDCODED * 1e6:
         # workaround for some kind of a bug which comes up with python 3.4
         # see https://github.com/datalad/datalad/issues/3150
+        # or on older CentOS with conda and python as new as 3.9
+        # see https://github.com/datalad/datalad/issues/5943
+        # TODO: let Yarik know that the world is a paradise now whenever 1e6
+        # is not large enough
         CMD_MAX_ARG = min(CMD_MAX_ARG, CMD_MAX_ARG_HARDCODED)
 except Exception as exc:
     # ATM (20181005) SC_ARG_MAX available only on POSIX systems

